### PR TITLE
Html anchor element

### PR DESCRIPTION
**Pull request by Indomian**
Added support for HTMLAnchorElement (<a>).

### CODE REVIEW

File: src/HTMLElement/elements/HTMLAnchorElement.js [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+    setAttribute(key, value) {
+        if (key === 'href') {
+            this.href = value;
+        } else {
+            super.setAttribute(key, value);
+        }
+    }
+
+    getAttribute(key) {
+        if (key === 'href') {
```
Comment (christophehurpeau) [inline src/HTMLElement/elements/HTMLAnchorElement.js]: why ?

Comment (Indomian) [inline src/HTMLElement/elements/HTMLAnchorElement.js]: Ehh, to provide standart behaviour. Or i missed something? But on second look seems i'm wrong.
I was trying to provide normal behaviour for such situation:
```javascript
let object = new HTMLAnchorElement();
object.setAttribute('href', '/some_relative_url');
assert.equal(object.href, 'http://full.url/some_relative_url'); //True, cause we updated inner _href field.
```

Comment (christophehurpeau) [inline src/HTMLElement/elements/HTMLAnchorElement.js]: Tried on my browser:
```
let object = document.createElement('a');
> undefined
object.setAttribute('href', '/some_relative_url');
> undefined
object.href
> "https://github.com/some_relative_url"
object.getAttribute('href')
> "/some_relative_url"
```
So getAttribute returns the setted attribute

Comment (Indomian) [inline src/HTMLElement/elements/HTMLAnchorElement.js]: Ok, seems first half is right. So we need to updated setter only. Ok i will fix this.

**Comment by Indomian** [comment]
Removed wiered behaviour in getAttribute.

File: src/HTMLElement/elements/HTMLAnchorElement.js [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+export default class HTMLAnchorElement extends HTMLElement {
+    constructor() {
+        super();
+        this._href = new Url();
+    }
+    /**
+     * @inheritDoc
+     */
+    setAttribute(key, value) {
+        super.setAttribute(key, value);
```
Comment (christophehurpeau) [inline src/HTMLElement/elements/HTMLAnchorElement.js]: If href setter calls setAttribute, why call it here too?

File: src/HTMLElement/elements/HTMLAnchorElement.js [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+     */
+    get hash() {
+        return this._href.hash;
+    }
+
+    /**
+     * @param {string} value
+     */
+    set hash(value) {
+        this._href.hash = value;
```
Comment (christophehurpeau) [inline src/HTMLElement/elements/HTMLAnchorElement.js]: Does this should also change the href attribute?

**Comment by christophehurpeau** [comment]
Thanks!

File: src/HTMLElement/elements/HTMLAnchorElement.js (hunk 2 above)
Comment (Indomian) [inline src/HTMLElement/elements/HTMLAnchorElement.js]: This is for that test:
```javascript
let object = document.createElement('a');
> undefined
object.setAttribute('href', '/some_relative_url');
> undefined
object.href
> "https://github.com/some_relative_url"
object.getAttribute('href')
> "/some_relative_url"

```

so when we set it through attribute it's changed in href too.

File: src/HTMLElement/elements/HTMLAnchorElement.js (hunk 3 above)
Comment (Indomian) [inline src/HTMLElement/elements/HTMLAnchorElement.js]: Seems no. Can you sujjest case for this? We can left this this way until we need other behaviour.

Comment (christophehurpeau) [inline src/HTMLElement/elements/HTMLAnchorElement.js]: screen shot 2015-12-23 at 20 43 02

Comment (Indomian) [inline src/HTMLElement/elements/HTMLAnchorElement.js]: Here is some more representative:

```javascript
var a=document.createElement('a');
undefined
a.href='/';
"/"
a.href
"https://github.com/"
a.hash='#some'
"#some"
a.href
"https://github.com/#some"
a.getAttribute('href')
"https://github.com/#some"
```

So as i understand a.getAttribute behaves different ways when a.hash or something changed. It's still relative when you do a.href='/something', but it changes to absoulte when you change it with .hash and other fields. What we gona do? Lets left current behaviour as is.

File: src/HTMLElement/elements/HTMLAnchorElement.js (hunk 2 above)
Comment (christophehurpeau) [inline src/HTMLElement/elements/HTMLAnchorElement.js]: I can't find this test in the tests, did you implemented it ? According to your implementation, I think it's gonna fail

Comment (Indomian) [inline src/HTMLElement/elements/HTMLAnchorElement.js]: Seems it in external tests i'm using html-document for. I'll add test and check.

File: src/HTMLElement/elements/HTMLAnchorElement.js (hunk 3 above)
Comment (christophehurpeau) [inline src/HTMLElement/elements/HTMLAnchorElement.js]: Please, can you make this test pass ? You can find what to do here :

https://html.spec.whatwg.org/multipage/semantics.html#api-for-a-and-area-elements:dom-hyperlink-hash-3

File: src/HTMLElement/elements/HTMLAnchorElement.js (hunk 2 above)
Comment (Indomian) [inline src/HTMLElement/elements/HTMLAnchorElement.js]: I created such test:
```javascript
test('HTMLAnchorElement setAttribute', () => {
    let document = new Document();
    document.location.href = 'https://github.com/';
    let object = document.createElement('a');
    object.setAttribute('href', '/some_relative_url');
    expect(object.href, 'https://github.com/some_relative_url');
    expect(object.getAttribute('href'), '/some_relative_url');
});
```
And it's passing well. On current version of branch.

File: src/HTMLElement/elements/HTMLAnchorElement.js (hunk 3 above)
Comment (Indomian) [inline src/HTMLElement/elements/HTMLAnchorElement.js]: Done. Also updated protocol, path and other same setters.

Comment (christophehurpeau) [inline src/HTMLElement/elements/HTMLAnchorElement.js]: Great ! Thanks :)

Comment (christophehurpeau) [inline src/HTMLElement/elements/HTMLAnchorElement.js]: I can't find where you handle "javascript" scheme or empty/null value/url ?

Comment (Indomian) [inline src/HTMLElement/elements/HTMLAnchorElement.js]: Implemented as described.

**Comment by Indomian** [comment]
@christophehurpeau Hello, could you please merge this. Seems there are a lot of dependencies updates from last build of master. Hope we can continue working on library.

**Comment by Indomian** [comment]
Thank you!

**Pull request merged**